Makefile

```diff
@@ -35,32 +35,39 @@ install-html:
 lint: lint-black lint-isort lint-flake8 lint-t
 
 lint-black:
-	$(PYTHON) -m black --check --quiet --diff . stg
+	$(PYTHON) -m black --check --quiet --diff . stgit/main.py
 
 lint-isort:
-	$(PYTHON) -m isort --check-only --quiet --diff . stg
+	$(PYTHON) -m isort --check-only --quiet --diff . stgit/main.py
 
 lint-flake8:
-	$(PYTHON) -m flake8 . stg
+	$(PYTHON) -m flake8 . stgit/main.py
 
 lint-t:
 	$(MAKE) -C t test-lint
 
 .PHONY: lint lint-black lint-isort lint-flake8 lint-t
 
 format:
-	$(PYTHON) -m black . stg
-	$(PYTHON) -m isort --quiet . stg
+	$(PYTHON) -m black . stgit/main.py
+	$(PYTHON) -m isort --quiet . stgit/main.py
 
-test: build
+test: .install
 	$(MAKE) -C t all
 
+.install: build
+	pyver=`$(PYTHON) -c 'import sys; print(".".join(map(str, sys.version_info[:2])))'` && \
+	prefix="$(PWD)/inst/python$$pyver" && \
+	sitepackages="$$prefix/lib/python$$pyver/site-packages" && \
+	mkdir -p "$$sitepackages" && \
+	PYTHONPATH="$$sitepackages" $(PYTHON) setup.py develop --prefix="$$prefix"
```

Comment:
I'm not immediately comfortable with this approach. It's both different than StGit's existing non-setuptool strategy as well as being unconventional relative to how setuptools and entry-points are normally used.

In other setuptools projects that I work on, the pattern is to run `pip install -e` or `setup.py develop` to reify the entry points into runnable programs. This typically happens in the context of a virtualenv, but could also conceivably be done in the context of the per-user site-packages (PEP 370).

Creating this `inst` directory instead of either using a virtualenv or the user's site-packages is the part that seems unconventional to me--and perhaps unnecessarily so?

I'm a little fuzzy on whether setuptools is truly *needed* on Windows or whether it is a convenience versus the status quo? Is it not possible to install StGit correctly on Windows without this change?

Reply:
The problem on Windows is that you need the exe-wrapper.

I was trying to emulate existing behavior in the `Makefile`. Maybe the `.install` target is not necessary at all and `make test` can be run in a simpler way?